static/etc/fstab

```diff
@@ -0,0 +1,49 @@
+/run/mnt/base / none bind,x-initrd.mount,ro 0 0
```

Comment:
Previously the paths that are needed by a given base was encoded inside a gitven base, instead of initrd. Thus allowing changes to base paths as needed inside i.e. core22 to be self-contained inside core22 snap.

With this change it become more coupled, where for example a package change in core22 snap requires simultanious change in core-initrd.

Also consider that a given kernel (core-initrd) might be booting and rolling back across different core22 revisions in recovery mode & run mode.

This also makes it harder to boot classic systems, as most of these mounts must not be done when booting classic systems with for example cloudimg-rootfs feature.

Do you think we could keep this file inside core22 snap at all? I understand that it might be harder to change fstab dynammically, and change what it does.

However, I was under the impression that systemd in the initrd can read $(rootfs)/etc/fstab and do things as defined there, no? Meaning it should be possible to keep this inside core22 snap "by magic".

Reply:
I am not sure I understand your comment.

This PR is for core-base. The corresponding PR for core-initrd in snapcore/core-initrd#74. I have removed from core-initrd as much as possible to move it to core-base. Now only `sysroot.mount` and `sysroot-writable.mount ` are left in core-initrd. `/usr/lib/modules`, `/usr/lib/firmware`, `/boot/*`are moved to core-base, where they should be.

So the coupling is actually reduced by this change.

> However, I was under the impression that systemd in the initrd can read $(rootfs)/etc/fstab and do things as defined there, no? Meaning it should be possible to keep this inside core22 snap "by magic".

This is exactly what snapcore/core-initrd#74 does.

Reply:
I was confused indeed. I thought this PR was for core-initrd. This is great!